core/src/main/java/org/ehcache/config/builders/ConfigurationBuilder.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.config.builders;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.Configuration;
+import org.ehcache.config.FluentCacheConfigurationBuilder;
+import org.ehcache.config.FluentConfigurationBuilder;
+import org.ehcache.core.config.DefaultConfiguration;
+import org.ehcache.spi.service.ServiceConfiguration;
+import org.ehcache.spi.service.ServiceCreationConfiguration;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.unmodifiableCollection;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * The {@code ConfigurationBuilder} enables building {@link Configuration}s using a fluent style.
+ *
+ * @author Alex Snaps
+ */
+public class ConfigurationBuilder implements FluentConfigurationBuilder<ConfigurationBuilder> {
```

Comment:
This class is exposed to the outside -- should it have some javadoc?

Reply:
It probably should. The overriding methods now get their documentation from the interface, but I will add documentation for the other methods and the static builder method.

Reply:
Added javadoc.